packages/frontend/src/pages/Space.tsx

```diff
@@ -1,57 +1,163 @@
+    };
+
+    const renderUpdateSpaceModal = () => {
+        return (
+            updateSpace && (
+                <SpaceActionModal
+                    projectUuid={projectUuid}
+                    spaceUuid={space?.uuid}
+                    actionType={ActionType.UPDATE}
+                    title="Update space"
+                    confirmButtonLabel="Update"
+                    icon="folder-close"
+                    onClose={() => setUpdateSpace(false)}
+                />
+            )
+        );
+    };
+
+    const renderDeleteSpaceModal = () => {
+        return (
+            deleteSpace && (
+                <SpaceActionModal
+                    projectUuid={projectUuid}
+                    spaceUuid={space?.uuid}
+                    actionType={ActionType.DELETE}
+                    title="Delete space"
+                    confirmButtonLabel="Delete"
+                    confirmButtonIntent={Intent.DANGER}
+                    icon="folder-close"
+                    onSubmitForm={() => {
+                        if (location.pathname.includes(space?.uuid)) {
+                            //Redirect to home if we are on the space we are deleting
+                            history.push(`/projects/${projectUuid}/home`);
+                        }
+                    }}
+                    onClose={() => {
+                        setDeleteSpace(false);
+                    }}
+                />
+            )
+        );
+    };
+
     return (
-        <Page>
+        <Center my="md">
             <Helmet>
-                <title>{data?.name} - Lightdash</title>
+                <title>{space?.name} - Lightdash</title>
             </Helmet>
-
-            <SpacePanel space={data}></SpacePanel>
-        </Page>
+            <Stack spacing="xl" w={900}>
+                <Group position="apart" mt="xs">
+                    <PageBreadcrumbs
+                        items={[
+                            {
+                                href: `/projects/${projectUuid}/spaces`,
+                                title: 'Spaces',
+                            },
+                        ]}
+                        mt="xs"
+                    >
+                        {space.name}
+                    </PageBreadcrumbs>
+                    <Group spacing="xs">
+                        <Can
+                            I="manage"
+                            this={subject('Space', {
+                                organizationUuid: user.data?.organizationUuid,
+                                projectUuid,
+                            })}
+                        >
+                            <ShareSpaceModal
+                                space={space!}
+                                projectUuid={projectUuid}
+                            />
+                            {renderSpaceBrowserMenu()}
+                            {renderUpdateSpaceModal()}
+                            {renderDeleteSpaceModal()}
```

Comment:
not a big fan of render methods. any particular reason why we might need them? it's just a lot of back and forth when reading the code

Reply:
just thought it'd make things tidier but i can get rid of them